zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/job/ActivatableJobsPushTest.java

```diff
@@ -276,16 +277,21 @@ private void assertEventOrder(
   }
 
   private void assertActivatedJobsPushed(final Long jobKey, final int activationCount) {
-    final var activatedJobs = jobStream.getActivatedJobs();
-    assertThat(activatedJobs).hasSize(activationCount);
-    activatedJobs.forEach(
-        activatedJob -> {
-          assertThat(activatedJob.jobKey()).isEqualTo(jobKey);
-
-          final JobRecord jobRecord = activatedJob.jobRecord();
-          assertThat(jobRecord.getWorkerBuffer()).isEqualTo(worker);
-          assertThat(jobRecord.getVariables()).isEqualTo(variables);
-          assertThat(jobRecord.getTenantId()).isEqualTo(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
-        });
+    await("waiting for the expected number of jobs to be activated")
+        .atMost(Duration.ofMillis(300))
```

Comment:
❓ While this works right now, this might run into flakiness in the future again, right? We are still relying on some timing now, we just wait longer. I was wondering if we can't let this wait for something more reliable (either in this method or before we call this method). Usually, we wait for specific events on the log. That probably won't suffice here. But maybe we can wait for something more specific in the `jobStream` object that semantically tells us that the stream of jobs was updated.

Reply:
If we can't - which is fine - we should at least fall back to waiting times that correlate with what we use for the recording exporter when we wait for records on the log to appear.